GoogleSignIn/Tests/Unit/GIDProfileDataFetcherTest.m

```diff
@@ -0,0 +1,150 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+// Test module imports
+@import GoogleSignIn;
+
+#import "GoogleSignIn/Sources/GIDHTTPFetcher/Implementations/Fakes/GIDFakeHTTPFetcher.h"
+#import "GoogleSignIn/Sources/GIDProfileDataFetcher/Implementations/GIDProfileDataFetcher.h"
+#import "GoogleSignIn/Tests/Unit/OIDAuthState+Testing.h"
+#import "GoogleSignIn/Tests/Unit/OIDTokenResponse+Testing.h"
+
+static NSString *const kErrorDomain = @"ERROR_DOMAIN";
+static NSInteger const kErrorCode = 400;
+
+@interface GIDProfileDataFetcherTest : XCTestCase
+@end
+
+@implementation GIDProfileDataFetcherTest {
+  GIDProfileDataFetcher *_profileDataFetcher;
+  GIDFakeHTTPFetcher *_httpFetcher;
+}
+
+- (void)setUp {
+  [super setUp];
+  _httpFetcher = [[GIDFakeHTTPFetcher alloc] init];
+  _profileDataFetcher = [[GIDProfileDataFetcher alloc] initWithHTTPFetcher:_httpFetcher];
+}
+
+- (void)testFetchProfileData_outOfAuthState_success {
+  OIDTokenResponse *tokenResponse =
+      [OIDTokenResponse testInstanceWithIDToken:[OIDTokenResponse fatIDToken]
+                                    accessToken:kAccessToken
+                                      expiresIn:@(300)
+                                   refreshToken:kRefreshToken
+                                   tokenRequest:nil];
+  OIDAuthState *authState = [OIDAuthState testInstanceWithTokenResponse:tokenResponse];
+  
+  GIDHTTPFetcherTestBlock testBlock =
+      ^(NSURLRequest *request, GIDHTTPFetcherFakeResponseProviderBlock responseProvider) {
+        XCTFail(@"_httpFetcher should not be invoked");
+      };
+  [_httpFetcher setTestBlock:testBlock];
+  
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:@"Callback called with no error"];
```

Comment:
Technically, the expectation is only used to verify that the callback is invoked. So it'd be fine to use "Callback invoked" as the description.

Same for other places.

Reply:
Good catch.